java-function/DEPLOYING.md

```diff
@@ -14,7 +14,7 @@ In order to further develop this application the following tools may be needed:
 To quickly test locally, run this command in the root directory.
 
 ```
-pack build java-function --path . --builder ghcr.io/vmware-tanzu/function-buildpacks-for-knative/functions-builder:0.0.12
+pack build java-function --path . --builder ghcr.io/vmware-tanzu/function-buildpacks-for-knative/functions-builder:0.1.0 --env BP_FUNCTION=functions.Handler
```

Comment:
@joeeltgroth @andrew-su I noticed this `buildEnvs` field in the [func.yaml reference](https://github.com/knative/func/blob/main/docs/reference/func_yaml.md#buildenvs). At a glance, I don't believe our buildpacks support this but should we be? It could lead to us re-establishing a `func.yaml` here with that env baked in, instead of requiring it on a pack build.

Reply:
I gave it a try and it appears we do support buildEnv.  In other words, `pack build` without --env, but with a func.yaml with buildEnvs does result in a working image.  Were you thinking having func.yaml is closer to "works out of the box" and parallels the config/workload.yaml pattern better?

Reply:
We don't use the `buildEnvs` section because the building happens when you run the `pack` command (the `maven` or `gradle` buildpacks do the build and it happens before us). In the case of the func CLI it runs the pack command with the `buildEnvs` as envs. When it tries to deploy it'll put the `envs` as the service's `envs`.

Reply:
I wonder if we should document the subset of fields in `func.yaml` that we support, so that there's no confusion for why things like `buildEnvs` don't work

Reply:
Though I suppose it won't matter too much if folks are going to use the func CLI most of the time. I'm not sure what the right answer is in terms of how we would delineate between fields supported for pack builds vs func CLI builds.

Reply:
@joeeltgroth @andrew-su Another option to make this more friendly out of the box is to keep a minimal func.yaml, even if it has almost nothing in it. Then the buildpack can detect based off of that (falling back to the default function name) and not require BP_FUNCTION to be set. This would be nearly equivalent -- from a user perspective -- to using `buildEnvs` (if it were actually supported).

Reply:
We could also make that change in our buildpack repo samples. I tend to think detection via source code is more intuitive for users as it follows the "configuration is code" philosophy and is therefore more repeatable, with less risk of user error (i.e. typo in pack command).

Reply:
Discussed the pros and cons of having func.yaml during team meeting.  We'll leave func.yaml out of these accelerators (at least until further feedback) as the focus here should be on workload.yaml and TAP.